### PR TITLE
favor-state option support for --defer

### LINKED ISSUE #2021: Add favor-state feature to defer documentation
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/reference/node-selection/defer

### What part(s) of the page would you like to see updated?

Addition of documentation around the new favor-state feature which which enables --defer to favor using --state node even if node exists in current target.

This optionally removes the below second --defer criterion when using the --favor-state flag.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by matthewshaver**
changes to the defer feature

## What are you changing in this pull request and why?

Added changelog to address v1.4 change to the defer feature. 

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding new pages (delete if not applicable):
- [ ] Add page to `website/sidebars.js`
- [ ] Provide a unique filename for the new page

Removing or renaming existing pages (delete if not applicable):
- [ ] Remove page from `website/sidebars.js`
- [ ] Add an entry `_redirects`
- [ ] [Ran link testing](https://github.com/dbt-labs/docs.getdbt.com#running-the-cypress-tests-locally) to update the links that point to the deleted page

### CODE REVIEW

File: website/docs/reference/node-selection/defer.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ title: "Defer"
 
 </Changelog>
 
-**N.B.** Deferral is a powerful, complex feature that enables compelling workflows. We reserve the right to change the name and syntax in a future version of dbt to make the behavior clearer and more intuitive. For details, see [dbt#2968](https://github.com/dbt-labs/dbt-core/issues/2968).
+**N.B.** Deferral is a powerful, complex feature that enables compelling workflows. As the use cases for `--defer` evolve, enhancements may be made to the feature, but we are committed to providing backward compatibility for supported versions of dbt Core.  For details, see [dbt#5095](https://github.com/dbt-labs/dbt-core/discussions/5095).
```
Comment (runleonarun) [inline website/docs/reference/node-selection/defer.md]: Just making this active.
```suggestion
**N.B.** Deferral is a powerful, complex feature that enables compelling workflows. As the use cases for `--defer` evolve, we could make enhancements to the feature, but commit to providing backward compatibility for supported versions of dbt Core.  For details, see [dbt#5095](https://github.com/dbt-labs/dbt-core/discussions/5095).
```

Comment (runleonarun) [reply website/docs/reference/node-selection/defer.md]: What does N.B. mean?

File: website/docs/reference/node-selection/defer.md [hunk 2]
```diff
@@ -33,6 +33,12 @@ $ dbt test --models [...] --defer --state path/to/artifacts
 ```
 
 </VersionBlock>
+<Changelog>
+
+- **v1.4**: Added the `--favor-state` option, which enables `--defer` to favor using the `--state` node even if the node exists in the current target. Using the `--favor-state` flag renders number 2 in following the criteria optional.
```
Comment (runleonarun) [inline website/docs/reference/node-selection/defer.md]:
```suggestion
- **v1.4**: Added the `--favor-state` option, which enables `--defer` to favor using the `--state` node even if the node exists in the current target. Using the `--favor-state` flag renders number 2 in the following criteria optional.
```

**Review by runleonarun — COMMENTED** [review]
A tiny typo fix and and suggestion!

File: website/docs/reference/node-selection/defer.md [hunk 3]
```diff
@@ -9,7 +9,7 @@ title: "Defer"
 
 </Changelog>
 
-**N.B.** Deferral is a powerful, complex feature that enables compelling workflows. We reserve the right to change the name and syntax in a future version of dbt to make the behavior clearer and more intuitive. For details, see [dbt#2968](https://github.com/dbt-labs/dbt-core/issues/2968).
+Deferral is a powerful, complex feature that enables compelling workflows. As the use cases for `--defer` evolve, we could make enhancements to the feature, but commit to providing backward compatibility for supported versions of dbt Core.  For details, see [dbt#5095](https://github.com/dbt-labs/dbt-core/discussions/5095).
```
Comment (nghi-ly) [inline website/docs/reference/node-selection/defer.md]:
```suggestion
Deferral is a powerful, complex feature that enables compelling workflows. As the use cases for `--defer` evolve, dbt Labs could make enhancements to the feature, but commit to providing backward compatibility for supported versions of dbt Core.  For details, see [dbt#5095](https://github.com/dbt-labs/dbt-core/discussions/5095).
```

File: website/docs/reference/node-selection/defer.md [hunk 4]
```diff
@@ -33,6 +33,12 @@ $ dbt test --models [...] --defer --state path/to/artifacts
 ```
 
 </VersionBlock>
+<Changelog>
+
+- **v1.4**: Added the `--favor-state` option, which enables `--defer` to favor using the `--state` node even if the node exists in the current target. Using the `--favor-state` flag renders number 2 in the following criteria optional.
```
Comment (nghi-ly) [inline website/docs/reference/node-selection/defer.md]:
```suggestion
- **v1.4**: Added the `--favor-state` option, which enables `--defer` to favor using the `--state` node even if the node exists in the current target. Using the `--favor-state` option renders the second criteria in the following optional.
```

Comment (nghi-ly) [reply website/docs/reference/node-selection/defer.md]: - sugg for consistency for translation purposes
- sugg for maybe better readability

**Review by nghi-ly — COMMENTED** [review]
looks good! just a couple editorial comments

**Pull request merged**